Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -266,6 +264,8 @@ auto TreeResolver::resolveElement(Element& element, const RenderStyle* existingS
     auto update = createAnimatedElementUpdate(WTFMove(resolvedStyle), styleable, parent().change, resolutionContext);
 
     if (!affectsRenderedSubtree(element, *update.style)) {
+        if (update.style->display() == DisplayType::None && element.hasDisplayNone())
+            element.storeDisplayContentsOrNoneStyle(WTFMove(update.style));
```

Comment:
This is not correct. TreeResolver must not store the style as it is an operation equivalent to render tree building. Instead it the style should be added to Style::Update and storing should be done by the storeDisplayContentsOrNoneStyle call in RenderTreeUpdater.